### ISSUE TITLE
Allow next on step verification failed

### ISSUE THREAD

**Issue by francescopedronomnys** [stepstone-tech/android-material-stepper#162]
Hello,
I don't know if this feature is already available or not.
Is it possible to use steps verification (e.g. to show error icon when using in tabs mode) without enforcing steps completion in sequence?
My use case is that the user should be able to complete steps in any order and I would like to maintain the error icon on tab name if a step is invalid.

Thanks

**Comment by zawadz88** [comment]
Hi,
I don't think it's possible with the library currently.
However, TBH I don't think it should be possible anyway. 
The documentation says: "Steppers display progress through a sequence of logical and numbered steps."
Completing steps in any order seems like a misuse of the stepper component.
Could you describe your use case a bit more?

**Comment by francescopedronomnys** [comment]
If you look at the documentation at chapter "Non-linear steppers" it describes a use case similar to mine (without talking about validation).
I can confirm that the library does not allow it because in onNext() method you return as soon as the current step is not verified. I'll look if I'm able to modify it to include my use case.

What I'm implementing is a form with multiple pages that can be completed in any order. I would like also to show which pages are valid when user click on next button.

In any case thanks for the library.

**Comment by zawadz88** [comment]
Hi @francescopedronomnys,
Thanks for the response. I'm not sure about this, but I think when you rotate the device some information about failing steps might get lost.
This might be an issue if you want to always keep that info and use it for something more than an indication that the user cannot go further.

**Comment by francescopedronomnys** [comment]
Yeah, my solution is a quick hack, so it will have some issues for sure.
Hope to see that feature implemented in the future, I'll try to improve my implementation and let you know in case it works right.

**Comment by sacrificium-dev** [comment]
Same here. Would be nice if I could use Non-linear steppers and finish them in any order.
Anyway the library is great so far. Designed along Google's design guideline. Well done!
